File: src/main/java/com/aseubel/yusi/pojo/entity/SoulMessage.java
package com.aseubel.yusi.pojo.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.DynamicInsert;
import org.hibernate.annotations.DynamicUpdate;

import java.time.LocalDateTime;

/**
 * 匹配用户间的聊天消息
 */
@Data
@Entity
@Builder
@Table(name = "soul_message")
@DynamicInsert
@DynamicUpdate
@AllArgsConstructor
@NoArgsConstructor
public class SoulMessage {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "match_id")
    private Long matchId;

    @Column(name = "sender_id")
    private String senderId;

    @Column(name = "receiver_id")
    private String receiverId;

    @Column(name = "content", length = 1000)
    private String content;

    @Column(name = "is_read")
    private Boolean isRead;

    @Column(name = "create_time")
    private LocalDateTime createTime;
}

File: backend-go/internal/model/soul_message.go
package model

import "time"

type SoulMessage struct {
	ID         int64     `gorm:"primaryKey;autoIncrement" json:"id"`
	MatchID    int64     `gorm:"column:match_id;index" json:"matchId"`
	SenderID   string    `gorm:"column:sender_id;size:64;index" json:"senderId"`
	ReceiverID string    `gorm:"column:receiver_id;size:64;index" json:"receiverId"`
	Content    string    `gorm:"size:1000" json:"content"`
	IsRead     bool      `gorm:"column:is_read;default:false" json:"isRead"`
	CreateTime time.Time `gorm:"column:create_time;index" json:"createTime"`
}

func (SoulMessage) TableName() string {
	return "soul_message"
}
